src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -124,21 +124,75 @@ void CALLBACK cfApiFetchDataCallback(const CF_CALLBACK_INFO *callbackInfo, const
     }
 
     qint64 offset = 0;
+    QByteArray protrudingData;
+    bool protrudingDataLeft = false;
 
     QObject::connect(&socket, &QLocalSocket::readyRead, &loop, [&] {
-        auto data = socket.readAll();
+        const auto data = socket.readAll();
         if (data.isEmpty()) {
             qCWarning(lcCfApiWrapper) << "Unexpected empty data received" << requestId;
             sendTransferError();
             loop.quit();
             return;
         }
-        sendTransferInfo(data, offset);
-        offset += data.length();
+
+        QByteArray dataToSent;
+        // CFAPI expectes sended blocks to be of a specified size.
```

Comment:
@FlexW Send in past is just sen**t**. Sorry for being such a grammar smartass :)

Having said that, shouldn't we say that the data must be "multiple of a block size", not just - "specified block size"? Just to make sure we have a clear intent here and do not get confused when reading this somewhat later :)

Reply:
As said I tried it in the first place with multiple 4k blocks but that didn't work out.

Reply:
> As said I tried it in the first place with multiple 4k blocks but that didn't work out. That's why I choose this phrasing;)
@FlexW 
I wonder why it didn't work for you. Could be some other mistake in the logic.

Reply:
@allexzander okay I did again some tests. Just multiply the block size by two. For pictures with 41 KB it worked. It also worked for a video with 3.72 MB but then it failed for the Nextcloud installer with 84 MB. I don't know if it was random but yeah. Another thing I noticed, is that if I free up space with a right click on a folder and press free up local space I have then often (not always) problems when downloading the video and the installer again. Maybe you could also do some tests on your side.